autoware_lanelet2_extension/docs/extra_lanelet_subtypes.md

```diff
@@ -43,3 +43,29 @@ Sample pedestrian lane in .osm format is shown below:
     <tag k="one_way" v="yes"/>
   </relation>
 ```
+
+## Bicycle Lane (format_version >= 2)
+
+Although `bicycle_lane` subtype is already defined in the original [Lanelet2 documentation](https://github.com/fzi-forschungszentrum-informatik/Lanelet2/blob/master/lanelet2_core/doc/LaneletAndAreaTagging.md#subtype-and-location), Autoware compatible with `format_version>=2` supports `bicycle_lane` at component level.
+
+- refers: `road` lanelet adjacent to the objective `bicycle_lane` should share the same boundary LineString
+- specification:
+  - A `bicycle_lane` is connected to/from only `bicycle_lane` lanelets and is adjacent to a `road` lanelet. Namely in Left(Right) hand side traffic rules, it is on the left(right) side of `road` lanelet.
+  - A `road_shoulder` lanelet must not be adjacent to a `bicycle_lane` because pullover on `bicycle_lane` is prohibited. For simplifying planning/prediction component, at format_version 2, a `bicycle_lane` should cover the adjacent area corresponding to `road_shoulder` or `pedestrian_lane` and therefore should have `road_border` as either of its boundary.
```

Comment:
I am a bit against setting unnecessary limitations in map format to accommodate with Autoware's implementation. I feel that the map should reflect the actual physical indication of the road as much as possible, and make Autoware capable of handling them properly even if that means complicating some implementations. 

For example, there could be complex bicycle lanes where there could be a road next to bicycle lane as well:
https://www.hrr.mlit.go.jp/kanazawa/douro/bicycle/img/pdf/pdf05.pdf
![image](https://github.com/user-attachments/assets/82bccf2b-ea60-4ed6-a659-12274d655e70)

https://cyclelocker.net/nz_leftturn_lane/
![image](https://github.com/user-attachments/assets/bc04f195-deed-462c-9729-3a4be39d69e0)

Autoware's implementation does not have to capable of handling such scenes at this point, but the map format should have a way of representing such lane structure.